### ISSUE TITLE
Problems when installing WebApps.

### ISSUE THREAD

**Issue by perrsona1234** [vinifmor/bauh#56]
Hello.
When installing a Web-based application I'm encountering some issues.
I have tried to install two WebApps: Telegram & GMail.

Here are the logs:
Telegram:
https://pastebin.com/WdetT67P & https://pastebin.com/G7mf1R1X
GMail:
https://pastebin.com/0m2Tw4eY & https://pastebin.com/0FG62HBg

The OS is openSUSE Tumbleweed. I have followed the installation guide & turned it into a script for future reference, maybe I forgot about something?
Here it is: https://pastebin.com/cLzMYnWC

Should I provide anything else?
Thank You.

**Comment by vinifmor** [comment]
Hello, it seems you are running bauh with the root user. Theoretically it should not be a problem, but I'm going to test here to see if it happens to on my system as well. 

Could you give me a feedback if it works for a non-root user ?

**Comment by vinifmor** [comment]
I could reproduce a similar issue. First, I did a fresh bauh install and launched it within the root account. I was able to install Gmail and Instagram without issues. Then I started bauh as the default user. Tried to install Web applications with no success.

The issue was: as I firstly started bauh as the root user, it created the build folder on the temp dir as root. When the default user tries to use this folder, a permission denied is raised. I'm going to add a fix for this scenario.

**Comment by vinifmor** [comment]
I published some related fixes to the staging branch. Could you test it and give me feedback if it works now ? 

Before trying, execute the reset command:
```
bauh --reset 
```

To install the staging branch:
```
git clone https://github.com/vinifmor/bauh.git
cd bauh
git checkout staging
sudo pip3 install .
```

**Comment by perrsona1234** [comment]
So, I did reset bauh, installed the staging branch and tried installing Telegram again. It's not working.
Logs: https://pastebin.com/4nh27Mf6 & https://pastebin.com/y2y172sX

**Comment by vinifmor** [comment]
Ok... I'm going to install your distro and see if I can find out what's happening. In the meantime, try to install nativefier on your machine and set bauh to work with it.
You will have to install NodeJS / Npm first.  The just type on the terminal:
`
sudo npm install nativefier -g
`
To enable bauh to work with the system nativefier, got to **Settings** > **Web** and change **Nativefier** to **System**:

![system_nativefier](https://user-images.githubusercontent.com/4312414/74439266-b174f500-4e4a-11ea-9a39-6a444ac65d37.png)

Let me know if it works

**Comment by perrsona1234** [comment]
So...
I did install `npm` with `sudo zypper install --no-recommends npm` & then run `sudo npm install nativefier -g` to install `nativefier`. After running `bauh` as root, changing the `nativefier` to the system one and trying to install Gmail & Twitter - it worked. Both have been installed successfully & are running fine.

**Comment by perrsona1234** [comment]
To summarize everything for now...

AS A NORMAL USER:
Installing & running `flatpaks` is working
Installing & running `.appimages` is working
Installing & running `web apps` is working

AS A ROOT USER:
Installing & running `flatpaks` is working
Installing & running `.appimages` is working
Installing & running `web apps` is working after installing `bauh` from staging branch, installing `nativefier` & changing `bauh` settings to use `nativefier` provided by the OS.

Installing `snaps` works for both NORMAL & ROOT USERS, but when I try to run them they are showing this error:
`cannot perform readlinkat() on the mount namespace file descriptor of the init process: Permission denied`
(for 99.9% it's related to `snapd` itself and not `bauh`, because running snaps, which were installed via `sudo snap install` also causes this error).
`snaps` will probably work someday on openSUSE, when either repo's maintainter or upstream fixes them.

Do You want any more info?
Should I close the issue?
Thank You for all Your work.

**Comment by vinifmor** [comment]
@perrsona1234, you don't need to use the staging version to work with the system nativefier. It should work with 0.8.2. But I've already published today a new release ( 0.8.3 ) with the yesterday's staging code. 

Related to snaps, I'm not sure if you can run them as the root user. But I'm going to confirm this. Let the issue open for now until I test you scenario and see if it can be fixed.

**Comment by perrsona1234** [comment]
It was my mistake that `snaps` were not working. The repo I got `snapd` from was outdated. After following these instructions (https://snapcraft.io/install/mindustry/opensuse) & installing Mindustry with `sudo snap install mindustry` as a test - after executing `snap run mindustry` it worked fine.

But there is something broken in `bauh`. Installing snaps as a NORMAL & ROOT users works fine, but I can't run them (the button for that is grayed out). Running `snaps`, which I installed via `bauh` with `snap run` command works fine, tho.
